### PR TITLE
WIP Beginnings of a full tutorial.

### PR DESCRIPTION
**Pull request by tmcgilchrist**

### CODE REVIEW

**Comment by erikd-ambiata** [comment]
Err, isn't this the same as my renaming of "README" to "README.md" ? From the PR, I can't tell if this has any more actual content.

**Comment by tmcgilchrist** [comment]
There's slightly more content, but really it's so I actually finish it this time.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+newtype TweetId = TweetId {
+    unTweetId :: Text
+  } deriving (Eq, Show, Ord)
+
+```
+
+The automatic deriving of typeclasses is required by `Data.Map`, which needs `Eq` and `Ord`, while the `Show` is a convenience for us.
+
+Next we introduce the `Tweet` type which includes a number of fields and also defines `Eq` and `Show`.
+This time however we've decided to define our own `Ord` instance because we would like the Tweets to be ordered by their `tweetTime` field rather than using the default derived instance of `Ord`. Aside if you're interested in how derving works read this. [Link to how that works]()
```
Comment (thumphries) [inline README.md]: `s/derving/deriving/`

**Comment by thumphries** [comment]
nice to see more content, looks good to me 🍔

**Comment by erikd-ambiata** [comment]
:+1:

**Pull request closed without merging**